### PR TITLE
Add Config and Migration, Update Adapter.php

### PR DESCRIPTION
**Pull request by aphraoh**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

**Comment by aphraoh** [comment]
@vjik could you please merge this one too. Thanx

File: composer.json [hunk 1]
```diff
@@ -34,6 +34,8 @@
     "require": {
         "php": "^8.1",
         "yiisoft/db": "^1.0",
+        "yiisoft/db-migration": "^1.0",
```
Comment (xepozz) [inline composer.json]: Let's move it to the "suggests" section

File: composer.json [hunk 2]
```diff
@@ -34,6 +34,8 @@
...
+        "yiisoft/mutex-file": "^1.0",
```
Comment (xepozz) [inline composer.json]: Use `yiisoft/mutext` instead

File: src/Adapter.php [hunk 3]
```diff
@@ -33,16 +38,12 @@ public function __construct(
         private ConnectionInterface $db,
         private string $channel = QueueFactory::DEFAULT_CHANNEL_NAME,
     ) {
+        $this->mutex = new FileMutex(__CLASS__ . $this->channel, sys_get_temp_dir());
```
Comment (xepozz) [inline src/Adapter.php]: Use https://github.com/yiisoft/mutex/blob/master/src/MutexFactoryInterface.php#L19 instead

File: src/Adapter.php [hunk 4]
```diff
@@ -103,7 +104,8 @@ public function withChannel(string $channel): self
 
         $new = clone $this;
         $new->channel = $channel;
-
+        $new->mutex = new FileMutex(__CLASS__ . $this->channel, sys_get_temp_dir());
```
Comment (xepozz) [inline src/Adapter.php]: Use https://github.com/yiisoft/mutex/blob/master/src/MutexFactoryInterface.php#L19 instead

File: src/Adapter.php [hunk 5]
```diff
@@ -194,4 +196,25 @@ private function moveExpired(): void
...
+        while (1) {  // TWK TODO while condition should not be hard coded, use LoopInterface
+            if ($payload = $this->reserve()) {
+                if ($handlerCallback(\unserialize($payload['job']))) {
+                    $this->release($payload);
+                }
+            } elseif (!$repeat) {
+                break;
+            } elseif ($timeout) {
+                sleep($timeout);
+            }
```
Comment (xepozz) [inline src/Adapter.php]:
```suggestion
                continue;
            } 
            if (!$repeat) {
                break;
            } 
            if ($timeout > 0) {
                sleep($timeout);
            }
```

File: src/Adapter.php [hunk 6]
```diff
@@ -194,4 +196,25 @@ private function moveExpired(): void
...
+    /**
+     * Listens queue and runs each job.
+     *
+     * @param callable(MessageInterface): bool  $handlerCallback The handler which will handle messages. Returns false if it cannot continue handling messages
+     * @param bool $repeat whether to continue listening when queue is empty.
+     * @param non-negative-int $timeout number of seconds to sleep before next iteration.
+     */
+    public function run(callable $handlerCallback, bool $repeat, int $timeout = 0): void
+    { 
+        while (1) {  // TWK TODO while condition should not be hard coded, use LoopInterface
```
Comment (xepozz) [inline src/Adapter.php]: Why wouldn't you use it now?

**Review by xepozz — CHANGES_REQUESTED** [review]

File: config/di.php
Comment (vjik) [inline config/di.php:1]: You can remove configuration with empty array.

Comment (aphraoh) [reply config/di.php:1]: Hi @vjik, when I do that there is issues either with composer update or something else, I can not remember what it was or why, can we leave it for now?

**Pull request closed without merging**